src/peergos/gwt/emu/java/nio/file/Path.java

```diff
@@ -35,7 +35,16 @@ public Path getParent() {
     }
 
     public Path getFileName() {
-        throw new IllegalArgumentException("Not implemented!");
+        int index = pathString.lastIndexOf(SEPARATOR);
+        if (index == -1) {
+            return new Path(pathString);
+        } else {
+            if(index == pathString.length() - 1) {
+                return null;
```

Comment:
why null?

Reply:
That is what the Path method does also.

Reply:
It only returns null if the path it empty: https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#getFileName()

We are different on e.g. /somedir/